etc/scripts/sync-configured-state-from-csv.py

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env python
+
+"""
+Script to set isConfigured value against Jira App properties from CSV of JiraHosts.
+
+Takes an installationIds.csv with installationIds.
+Keeps track of processed entries in a separate file, so that script can be stopped
+and resumed without having to change the input file.
+
+Input file format:
+    installationId
+    12345
+
+Running the script:
+    $ python3 ./sync-configured-state-from-csv.py --env [ local | dev | staging | prod ] --sleep [ sleep-duration ] --input [ input-file-name.csv ] --output [ output-file-name.csv ]
+Example
+    $ python3 ./sync-configured-state-from-csv.py --env local --sleep 10 --input installationIds-current.csv --output configurationOutput.csv
+
+First time setup:
+
+1. virtualenv env -p python3
+2. source env/bin/activate
+3. pip install requests
+"""
+
+import argparse
+import csv
+import logging
+import requests
+import subprocess
+import sys
+import time
+import json
+from dataclasses import dataclass
+from requests import Request
+from requests.auth import AuthBase
+from typing import NamedTuple
+
+LOG = logging.getLogger(__name__)
+
+session = requests.Session()
+
+
+def configure_logging(level='DEBUG'):
+    formatter = logging.Formatter(
+        '%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+
+    ch = logging.StreamHandler(sys.stdout)
+    ch.setFormatter(formatter)
+    ch.setLevel(level)
+
+    root = logging.getLogger()
+    root.setLevel(level)
+    root.addHandler(ch)
+
+
+class SLAuth(AuthBase):
+    def __init__(self, audience, group, environment):
+        self.audience = audience
+        self.group = group
+        self.environment = environment
+
+    def __call__(self, req: Request) -> Request:
+        cmd = ['atlas',
+               'slauth', 'token',
+               '-m',
+               '-g', self.group,
+               '-e', self.environment,
+               '-a', self.audience]
+        data = subprocess.check_output(cmd)
+        jwt = data.decode('utf-8').strip()
+
+        req.headers['Authorization'] = f'slauth {jwt}'
+        return req
+
+
+def create_slauth(env: str) -> SLAuth:
+    return SLAuth(
+        audience=f'github-for-jira',
+        group=f'micros-sv--github-for-jira-dl-admins',
+        environment=env
+    )
+
+
+class Environment(NamedTuple):
+    github_for_jira_url: str
+    github_for_jira_auth: object
+
+
+def create_environment(env: str) -> Environment:
+    print("Running for env", env)
+    if env == 'dev':
+        return Environment(
+            github_for_jira_url='https://jkay-tunnel.public.atlastunnel.com',
```

Comment:
This hard-coded value might need to be pulled from the `env.local`.

Reply:
ooooo good catch!!! that was me testing.